File: Python/Logic-2/make_chocolate.py
def make_chocolate(small, big, goal):
    if goal > small + (big * 5) or small < goal % 5:
        return -1
    elif goal // 5 <= big:
        if goal // 5 <= big and small >= goal % 5:
            return goal % 5
    else:
        return goal - big * 5

File: Java/Logic-2/makeChocolate.java
public int makeChocolate(int small, int big, int goal) {
    if (goal > small + big * 5 || goal - Math.min(big, goal / 5) * 5 > small) {
        return -1;
    }
    return goal - Math.min(big, goal / 5) * 5;
}